### PR TITLE
Don't require `defaultTheme`, `defaultIconTheme` and `applicationName` in `FrontendApplicationConfig`

### PR DESCRIPTION
**Pull request by quantumsheep**
#### What it does
Remove the requirement for properties `defaultTheme`, `defaultIconTheme` and `applicationName` in `FrontendApplicationConfig` class.

#### How to test
Using `FrontendApplicationConfigProvider.set` we were required to set a value for the edited properties even though they have default values.

#### Review checklist

- [x] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

**Review by vince-fugnitto — COMMENTED** [review]
@quantumsheep thank you for the contribution, the changes seems to have caused build errors:
- https://github.com/eclipse-theia/theia/pull/9490/checks?check_run_id=2601093521#step:5:76
- https://github.com/eclipse-theia/theia/pull/9490/checks?check_run_id=2601093521#step:5:78
- https://github.com/eclipse-theia/theia/pull/9490/checks?check_run_id=2601093521#step:5:80

Please be sure to rebuild the whole project, it is possible that there are other errors besides those under `@theia/core`.

**Comment by quantumsheep** [comment]
@vince-fugnitto It should be good right now - I fixed default values and typing :)

**Comment by quantumsheep** [comment]
It seems like the windows build has failed because of a timeout issue. It might not be related to the content of this PR but to GitHub Actions servers being slow.

https://github.com/eclipse-theia/theia/pull/9490/checks?check_run_id=2602515183

File: dev-packages/application-package/src/application-props.ts [hunk 1]
```diff
@@ -36,6 +36,50 @@ export namespace NpmRegistryProps {
...
+    /**
+     * The name of the application. `Eclipse Theia` by default.
+     */
+    readonly applicationName?: string;
+
+    /**
+     * Electron specific configuration.
+     */
+    readonly electron?: Readonly<ElectronFrontendApplicationConfig>;
+}
```
Comment (vince-fugnitto) [inline dev-packages/application-package/src/application-props.ts:71]: I believe the following changes are unnecessary, we would only want to make the properties optional, not move the interfaces around in the same file.

Comment (quantumsheep) [reply dev-packages/application-package/src/application-props.ts]: I moved the interface because it made more sense to have it defined before being used.

Fixed in the last commit.

File: dev-packages/application-package/src/application-props.ts [hunk 2]
```diff
@@ -110,24 +106,34 @@ export interface FrontendApplicationConfig extends ApplicationConfig {
...
+type PickRequired<T, K extends keyof T> = Omit<T, K> & Required<Pick<T, K>>;
+
+export namespace FrontendApplicationConfig {
+    export const DEFAULT: PickRequired<FrontendApplicationConfig, 'applicationName' | 'defaultTheme' | 'defaultIconTheme'> = {
+        applicationName: 'Eclipse Theia',
+        defaultTheme: 'dark',
+        defaultIconTheme: 'none'
+    };
+}
+
```
[the remark this answers is not in the document]
Comment (quantumsheep) [reply dev-packages/application-package/src/application-props.ts:136]: Extending the design to other configurations makes sense however I'm not sure about predefined `required` and `partial` types, it seems like too much for me. What do you think?

[the remark this answers is not in the document]
Comment (marechal-p) [reply dev-packages/application-package/src/application-props.ts:136]: IIUC with your changes we already have two versions of the interfaces:

- `partial` is `FrontendApplicationConfig`.
- `resolved` is `FrontendApplicationConfig & typeof FrontendApplicationConfig.DEFAULT`.

Where `typeof FrontendApplicationConfig.DEFAULT` is a shortcut based on a runtime value that expands to `PickRequired<FrontendApplicationConfig, 'applicationName' | 'defaultTheme' | 'defaultIconTheme'>`.

So I thought it would be preferable to somehow get both interfaces as their own unique reference. Also note that on my branch, `FrontendApplicationConfig` still maps to an interface where fields with defaults are still required, only the `FrontendApplicationConfigProvider.set(...)` function now uses a more lax type.

Maybe you can just explicitly define both types so we can reference them without using `typeof SomeDefaultValue`, I think I'd be fine with just that. To define and export both `FrontendApplicationConfig` and `PartialFrontendApplicationConfig`.

[the remark this answers is not in the document]
Comment (quantumsheep) [reply dev-packages/application-package/src/application-props.ts:136]: Reading again the code I realized that the only issue I had with your modifications was the naming haha. Making the first letter as uppercase would be nicer for a type.

TLDR; I'm ok with your modifications

[the remark this answers is not in the document]
Comment (marechal-p) [reply dev-packages/application-package/src/application-props.ts:136]: Please update your PR at will. My modifications were for illustration purpose, all that's required is to expose both interfaces separately, no matter how you prefer to do it :)

[the remark this answers is not in the document]
Comment (quantumsheep) [reply dev-packages/application-package/src/application-props.ts:136]: I just understood why you named the properties starting with a lowercase. I don't know why but I though we could do `ConfigHelper<A, B>.Partial` haha. We can take your commit instead of mines, what do you think?

[the remark this answers is not in the document]
Comment (marechal-p) [reply dev-packages/application-package/src/application-props.ts:136]: https://github.com/eclipse-theia/theia/pull/9503

[the remark this answers is not in the document]
Comment (quantumsheep) [reply dev-packages/application-package/src/application-props.ts:136]: Perfect, I'm closing the current PR :)
